File: src/main/java/no/nav/pensjon/innsyn/domain/Beholdning.java
package no.nav.pensjon.innsyn.domain;

public class Beholdning {

    private final String datoFom;
    private final String datoTom;
    private final double belop;
    private final String beholdningstype;
    private final String status;
    private final double grunnlag;
    private final double grunnlagAvkortet;
    private final double innskudd;
    private final int inntektsar;
    private final double inntektsgrunnlag;
    private final int forstegangstjenesteAr;
    private final double ordinareDagpenger;
    private final int dagpengerAr;
    private final double dagpengerFisker;
    private final int omsorgAr;
    private final double omsorgBelop;
    private final double omsorgInnskudd;
    private final double uforeBelop;
    private final int uforeAr;
    private final double uforegrad;
    private final double uforeYrkesskadegrad;
    private final double uforeAntattInntektYrke;
    private final boolean uforeYrkesskade;
    private final boolean uforeUforetrygd;
    private final boolean uforeUforeAr;
    private final double uforeAntattInntekt;
    private final double reguleringBelop;
    private final String reguleringDato;

    public Beholdning(String datoFom,
                      String datoTom,
                      double belop,
                      String beholdningstype,
                      String status,
                      double grunnlag,
                      double grunnlagAvkortet,
                      double innskudd,
                      int inntektsar,
                      double inntektsgrunnlag,
                      int forstegangstjenesteAr,
                      double ordinareDagpenger,
                      int dagpengerAr,
                      double dagpengerFisker,
                      int omsorgAr,
                      double omsorgBelop,
                      double omsorgInnskudd,
                      double uforeBelop,
                      int uforeAr,
                      double uforegrad,
                      double uforeYrkesskadegrad,
                      double uforeAntattInntektYrke,
                      boolean uforeYrkesskade,
                      boolean uforeUforetrygd,
                      boolean uforeUforeAr,
                      double uforeAntattInntekt,
                      double reguleringBelop,
                      String reguleringDato) {
        this.datoFom = datoFom;
        this.datoTom = datoTom;
        this.belop = belop;
        this.beholdningstype = beholdningstype;
        this.status = status;
        this.grunnlag = grunnlag;
        this.grunnlagAvkortet = grunnlagAvkortet;
        this.innskudd = innskudd;
        this.inntektsar = inntektsar;
        this.inntektsgrunnlag = inntektsgrunnlag;
        this.forstegangstjenesteAr = forstegangstjenesteAr;
        this.ordinareDagpenger = ordinareDagpenger;
        this.dagpengerAr = dagpengerAr;
        this.dagpengerFisker = dagpengerFisker;
        this.omsorgAr = omsorgAr;
        this.omsorgBelop = omsorgBelop;
        this.omsorgInnskudd = omsorgInnskudd;
        this.uforeBelop = uforeBelop;
        this.uforeAr = uforeAr;
        this.uforegrad = uforegrad;
        this.uforeYrkesskadegrad = uforeYrkesskadegrad;
        this.uforeAntattInntektYrke = uforeAntattInntektYrke;
        this.uforeYrkesskade = uforeYrkesskade;
        this.uforeUforetrygd = uforeUforetrygd;
        this.uforeUforeAr = uforeUforeAr;
        this.uforeAntattInntekt = uforeAntattInntekt;
        this.reguleringBelop = reguleringBelop;
        this.reguleringDato = reguleringDato;
    }

    public String getDatoFom() {
        return datoFom;
    }

    public String getDatoTom() {
        return datoTom;
    }

    public double getBelop() {
        return belop;
    }

    public String getBeholdningstype() {
        return beholdningstype;
    }

    public String getStatus() {
        return status;
    }

    public double getGrunnlag() {
        return grunnlag;
    }

    public double getGrunnlagAvkortet() {
        return grunnlagAvkortet;
    }

    public double getInnskudd() {
        return innskudd;
    }

    public int getInntektsar() {
        return inntektsar;
    }

    public double getInntektsgrunnlag() {
        return inntektsgrunnlag;
    }

    public int getForstegangstjenesteAr() {
        return forstegangstjenesteAr;
    }

    public double getOrdinareDagpenger() {
        return ordinareDagpenger;
    }

    public int getDagpengerAr() {
        return dagpengerAr;
    }

    public double getDagpengerFisker() {
        return dagpengerFisker;
    }

    public int getOmsorgAr() {
        return omsorgAr;
    }

    public double getOmsorgBelop() {
        return omsorgBelop;
    }

    public double getOmsorgInnskudd() {
        return omsorgInnskudd;
    }

    public double getUforeBelop() {
        return uforeBelop;
    }

    public int getUforeAr() {
        return uforeAr;
    }

    public double getUforegrad() {
        return uforegrad;
    }

    public double getUforeYrkesskadegrad() {
        return uforeYrkesskadegrad;
    }

    public double getUforeAntattInntektYrke() {
        return uforeAntattInntektYrke;
    }

    public boolean getUforeYrkesskade() {
        return uforeYrkesskade;
    }

    public boolean getUforeUforetrygd() {
        return uforeUforetrygd;
    }

    public boolean getUforeUforeAr() {
        return uforeUforeAr;
    }

    public double getUforeAntattInntekt() {
        return uforeAntattInntekt;
    }

    public double getReguleringBelop() {
        return reguleringBelop;
    }

    public String getReguleringDato() {
        return reguleringDato;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/Beholdning.kt
package no.nav.pensjon.innsyn.domain

import javax.persistence.*

@Entity
@Table(name = "T_BEHOLDNING")
@SecondaryTables(
        SecondaryTable(name = "T_K_BEHOLDNING_T", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_BEHOLDNING_T")]),
        SecondaryTable(name = "T_K_BEHOLDNING_S", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_BEHOLDNING_S")]),
        SecondaryTable(name = "T_INNTEKT_OPPTJ", pkJoinColumns = [PrimaryKeyJoinColumn(name = "INNTEKT_OPPTJ_ID")]),
        SecondaryTable(name = "T_F_TJEN_OPPTJ", pkJoinColumns = [PrimaryKeyJoinColumn(name = "F_TJEN_OPPTJ_ID")]),
        SecondaryTable(name = "T_DAGPENGER_OPPTJ", pkJoinColumns = [PrimaryKeyJoinColumn(name = "DAGPENGER_OPPTJ_ID")]),
        SecondaryTable(name = "T_OMSORG_OPPTJ", pkJoinColumns = [PrimaryKeyJoinColumn(name = "OMSORG_OPPTJ_ID")]),
        SecondaryTable(name = "T_UFORE_OPPTJ", pkJoinColumns = [PrimaryKeyJoinColumn(name = "UFORE_OPPTJ_ID")]),
        SecondaryTable(name = "T_LONN_VEKST_REG", pkJoinColumns = [PrimaryKeyJoinColumn(name = "LONNSVEKST_REG_ID")])
)
data class Beholdning(
        @Column(name = "DATO_FOM")
        val datoFom: String,
        @Column(name = "DATO_TOM")
        val datoTom: String,
        @Column(name = "BELOP")
        val belop: Double,
        @Column(name = "DEKODE", table = "T_K_BEHOLDNING_T")
        val beholdningstype: String,
        @Column(name = "DEKODE", table = "T_K_BEHOLDNING_S")
        val status: String,
        @Column(name = "BEH_GRLAG")
        val grunnlag: Double,
        @Column(name = "BEH_GRLAG_AVKORTET")
        val grunnlagAvkortet: Double,
        @Column(name = "BEH_INNSKUDD")
        val innskudd: Double,
        @Column(name = "AR", table = "T_INNTEKT_OPPTJ")
        val inntektsar: Int,
        @Column(name = "BELOP", table = "T_INNTEKT_OPPTJ")
        val inntektsgrunnlag: Double,
        @Column(name = "AR", table = "T_F_TJEN_OPPTJ")
        val forstegangstjenesteAr: Int,
        @Column(name = "BELOP_ORDINAR", table = "T_DAGPENGER_OPPTJ")
        val ordinareDagpenger: Double,
        @Column(name = "AR", table = "T_DAGPENGER_OPPTJ")
        val dagpengerAr: Int,
        @Column(name = "BELOP_FISKERE", table = "T_DAGPENGER_OPPTJ")
        val dagpengerFisker: Double,
        @Column(name = "AR", table = "T_OMSORG_OPPTJ")
        val omsorgAr: Int,
        @Column(name = "BELOP", table = "T_OMSORG_OPPTJ")
        val omsorgBelop: Double,
        @Column(name = "OMS_OPPTJ_INNSKUDD", table = "T_OMSORG_OPPTJ")
        val omsorgInnskudd: Double,
        @Column(name = "BELOP", table = "T_UFORE_OPPTJ")
        val uforeBelop: Double,
        @Column(name = "AR", table = "T_UFORE_OPPTJ")
        val uforeAr: Int,
        @Column(name = "UFG", table = "T_UFORE_OPPTJ")
        val uforegrad: Double,
        @Column(name = "YUG", table = "T_UFORE_OPPTJ")
        val uforeYrkesskadegrad: Double,
        @Column(name = "ANNTATT_INNTEKT_YRKE", table = "T_UFORE_OPPTJ")
        val uforeAntattInntektYrke: Double,
        @Column(name = "YRKESSKADE", table = "T_UFORE_OPPTJ", columnDefinition = "INT(1)")
        val uforeYrkesskade: Boolean,
        @Column(name = "UFORETRYGD", table = "T_UFORE_OPPTJ", columnDefinition = "INT(1)")
        val uforeUforetrygd: Boolean,
        @Column(name = "UFOREAR", table = "T_UFORE_OPPTJ", columnDefinition = "INT(1)")
        val uforeUforeAr: Boolean,
        @Column(name = "ANNTATT_INNTEKT", table = "T_UFORE_OPPTJ")
        val uforeAntattInntekt: Double,
        @Column(name = "BELOP", table = "T_LONN_VEKST_REG")
        val reguleringBelop: Double,
        @Column(name = "REGULERING_DATO", table = "T_LONN_VEKST_REG")
        val reguleringDato: String
) {
    @Id
    @Column(name = "PERSON_ID")
    var personId: Int? = null
}